dev/tasks/conda-recipes/arrow-cpp/bld.bat

```diff
@@ -21,6 +21,7 @@ cmake -G "Ninja" ^
       -DARROW_PYTHON:BOOL=ON ^
       -DARROW_MIMALLOC:BOOL=ON ^
       -DARROW_FLIGHT:BOOL=ON ^
+      -DARROW_HDFS:BOOL=ON ^
```

Comment:
Note this isn't needed

https://github.com/apache/arrow/blob/master/cpp/CMakeLists.txt#L289

Is `pyarrow._hdfs` optional now?

Reply:
We have two filesystem implementations, the old python one and the new C++ implementation. `_hdfs` and `_s3` are the bindings for the C++ implementation. `pyarrow.hdfs` is not optional but `pyarrow._hdfs` (exposed under `pyarrow.fs`) is optional.

Reply:
`-DARROW_PYTHON:BOOL=ON` enables `ARROW_HDFS` automatically. So we don't need to specify `-DARROW_HDFS:BOOL=ON` explicitly.

Reply:
I prefer to be explicit about these options.